Gems/Terrain/Code/Source/TerrainSystem/TerrainSystem.h

```diff
@@ -236,5 +318,130 @@ namespace Terrain
+        for (int32_t i = 0; i < numJobs; ++i)
+        {
+            // If the number of positions can't be divided evenly by the number of jobs,
+            // ensure we still process the remaining positions along with the final job.
+            const size_t subSpanOffset = i * numPositionsPerJob;
+            const size_t subSpanCount = (i < numJobs - 1) ? numPositionsPerJob : AZStd::dynamic_extent;
+
+            // Define the job function using the sub span of positions to process.
+            const AZStd::span<VectorType>& positionsToProcess = inPositions.subspan(subSpanOffset, subSpanCount);
+            auto jobFunction = [=]()
+            {
+                // Process the sub span of positions, decrement the number of completions remaining,
+                // and invoke the completion callback if this happens to be the final job completed.
+                synchronousFunction(positionsToProcess, perPositionCallback, sampleFilter);
+                if (--(*numJobCompletionsRemaining) == 0 &&
+                    params && params->m_completionCallback &&
+                    !jobContext->IsCancelled())
+                {
+                    params->m_completionCallback();
+                }
+            };
+
+            // Create the job and start it immediately.
+            AZ::Job* processJob = AZ::CreateJobFunction(jobFunction, true, jobContext.get());
+            processJob->Start();
+        }
+
+        return jobContext;
+    }
+
+    template<typename SynchronousFunctionType>
+    inline AZStd::shared_ptr<AzFramework::Terrain::TerrainDataRequests::TerrainJobContext> TerrainSystem::ProcessFromRegionAsync(
+        SynchronousFunctionType synchronousFunction,
+        const AZ::Aabb& inRegion,
+        const AZ::Vector2& stepSize,
+        AzFramework::Terrain::SurfacePointRegionFillCallback perPositionCallback,
+        Sampler sampleFilter,
+        AZStd::shared_ptr<ProcessAsyncParams> params) const
+    {
+        // ToDo: Determine the number of jobs to split the work into based on:
+        // 1. The number of available worker threads.
+        // 2. The desired number of jobs as passed in.
+        // 3. The size of the area being processed.
+        //
+        // Note: We are currently restricting the number of worker threads to one
+        // because splitting the work over multiple threads causes contention when
+        // locking various mutexes, resulting in slower overall wall time for async
+        // requests split over multiple threads vs one where all the work is done on
+        // a single thread. The latter is still preferable over a regular synchronous
+        // call because it is just as quick and prevents the main thread from blocking.
+        // Once the mutex contention issues have been addressed, we should come up with
+        // an algorithm to break up 'inRegion' into sub-regions (or lists of positions?)
```

Comment:
This is an interesting problem. If split into regions, then you might want to send off intermediate callbacks that can say "this aabb region is done" so that way the user of the API could do something with the partial result (like upload data to the GPU).